src/Memory/ClientMemoryManager.ts

```diff
@@ -34,107 +34,109 @@ export class ReadOnlyClientMemoryManager {
         return match
     }
 
-    /**
-     * Get value of entity as a string
-     * @param entityName Name of Entity
-     */
-    public EntityValueAsString(entityName: string): string | null {
-        return this.curMemories.ValueAsString(entityName)
-    }
-
-    /**
-     * Get value of entity as string before most recent input
-     * @param entityName Name of Entity
-     */
-    public PrevEntityValueAsString(entityName: string): (string | null) {
-        return this.prevMemories.ValueAsString(entityName)
+    protected __ToString(value: string | number | boolean | object): string {
+        if (typeof value == 'object') {
+            return JSON.stringify(value);
+        }
+        return value.toString()
     }
 
     /**
-     * Get array of MemoryValues 
-     * @param entityName Name of Entity
+     * Get current value of entity 
      */
-    public EntityValues(entityName: string): MemoryValue[] {
-        return this.curMemories.Values(entityName)
+    public Get<T>(entityName: string, converter: (memoryValues: MemoryValue[]) => T): T {
```

Comment:
>converter: (memoryValues: MemoryValue[]) => T [](start = 38, length = 45)

better to make the converter optional and when it is null, by default use AS_VALUE_LIST converter.